src/ports/rankings/queries.ts

```diff
@@ -0,0 +1,251 @@
+import BN from 'bn.js'
+import SQL, { SQLStatement } from 'sql-template-strings'
+import {
+  getUniqueCollectorsFromCollectorsDayData,
+  getUniqueCreatorsFromCreatorsDayData,
+  getUniqueItemsFromItemsDayData
+} from '../../logic/rankings/utils'
+import {
+  RankingsFilters,
+  RankingsSortBy,
+  ItemsDayDataTimeframe,
+  RankingEntity,
+  RankingEntityResponse,
+  ItemRank,
+  CreatorRank,
+  CollectorRank,
+  RankingFragment,
+  ItemsDayDataFragment,
+  CollectorsDayDataFragment,
+  CreatorsDayDataFragment
+} from './types'
+
+export const MAX_RESULTS = 1000
+
+function getQueryParams(entity: RankingEntity, filters: RankingsFilters) {
+  const { from, category, rarity, sortBy } = filters
+  const conditions: SQLStatement[] = []
+
+  if (entity === RankingEntity.WEARABLES) {
+    conditions.push(SQL`search_emote_category IS NULL`)
+  } else if (entity === RankingEntity.EMOTES) {
+    conditions.push(SQL`search_wearable_category IS NULL`)
+  }
+
+  if (category) {
+    conditions.push(
+      entity === RankingEntity.WEARABLES ? SQL`search_wearable_category = ${category}` : SQL`search_emote_category = ${category}`
+    )
+  }
+
+  if (entity === RankingEntity.CREATORS) {
+    conditions.push(SQL`sales > 0`)
+    if (from === 0) {
+      conditions.push(SQL`collections > 0`)
+    }
+  } else if (entity === RankingEntity.COLLECTORS) {
+    conditions.push(SQL`purchases > 0`)
+  }
+
+  if (rarity) {
+    if (from === 0) {
+      // if it fetches the Item entity
+      conditions.push(SQL`rarity = ${rarity}`)
+    } else {
+      conditions.push(SQL`search_rarity = ${rarity}`)
+    }
+  }
+  if (from) {
+    conditions.push(SQL`date >= ${Math.round(from / 1000)}`)
+  }
+
+  let orderBy = 'volume'
+  let orderDirection = 'desc'
+  switch (sortBy) {
+    case RankingsSortBy.MOST_SALES:
+      if (entity === RankingEntity.COLLECTORS) {
+        orderBy = 'purchases'
+        orderDirection = 'desc'
+      } else if (from === 0 && entity === RankingEntity.CREATORS) {
+        orderBy = 'primary_sales'
+        orderDirection = 'desc'
+      } else {
+        orderBy = 'sales'
+        orderDirection = 'desc'
+      }
+      break
+    case RankingsSortBy.MOST_VOLUME:
+      if (entity === RankingEntity.COLLECTORS) {
+        // for accounts the field is "spent"
+        orderBy = 'spent'
+        orderDirection = 'desc'
+      } else if (entity === RankingEntity.CREATORS) {
+        // for accounts the field is "earned"
+        orderBy = from === 0 ? 'primary_sales_earned' : 'earned'
+        orderDirection = 'desc'
```

Comment:
is `orderDirection` always `desc`?

Reply:
mmm yes, let me change it for `const` then!